data-prepper-api/src/main/java/org/opensearch/dataprepper/model/annotations/AlsoRequires.java

```diff
@@ -0,0 +1,21 @@
+package org.opensearch.dataprepper.model.annotations;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Marker annotation used in schema generation to define the names and corresponding values of other required
+ * properties to be returned, if the property represented by the annotated field/method is present.
```

Comment:
```suggestion
 * Annotation used in schema generation to define the names and corresponding values of other required
 * configurations if the configuration represented by the annotated field/method is present.
```

I made this suggestion to generalize the documentation such that it is not just about schema generation. As I expressed in another comment, this should be an annotation to define constraints. We will use these for validations and schema generation.